lib/dependency_validator.dart

```diff
@@ -57,6 +63,29 @@ void run() {
 
   logger.info('Validating dependencies for $packageName\n');
 
+  // `loadPackageConfig` (used below) relies on having a file built from an absolute URI.
+  final packageConfig = File.fromUri(Uri.file('${Directory.current.path}/.dart_tool/package_config.json'));
+
+  // URIs in packageConfig may be relative to its directory. Change the working directory to account for this.
+  final rootDir = Directory.current;
+  final dir = Directory('${rootDir.path}/.dart_tool');
+  Directory.current = dir;
+
+  // Find packages that provide executables.
+  final packagesWithExecutables = Set<String>();
+  for (final package in (await loadPackageConfig(packageConfig)).packages) {
```

Comment:
Instead of checking every package (which includes transitive dependencies), we should move this block down below the code that parses the `pubspec.yaml` so that we can ignore the packages that aren't direct dependencies since they would never show up as "unused" anyway

Reply:
Since we are only reporting the intersection between this set and the set of packages explicitly depended on I'm not sure why it matters if there are extra packages in this set?

Reply:
Yeah, if we only report the intersection the end result will be the same, but it does mean we're doing work (checking the source of transitive deps) unnecessarily. Might not actually take that long to do since we're just parsing a couple file paths and checking to see if a directory exists, but `wdesk` for example has 260 total dependencies so it might add up.

Reply:
Eh.. looking at the code again, it's so simple that it's probably not worth changing this. Running in wdesk takes ~3.5s and in this repo it takes ~2s so there isn't much juice to squeeze there anyway.